package/utils/busybox/convert_defaults.pl

```diff
@@ -1,5 +1,106 @@
 #!/usr/bin/env perl
 
+sub default_for_applet($$$) {
+	my $skip_hash = shift;
+	my $applet = shift;
+	my $default = shift;
+	my @newdefault = ();
+
+	foreach my $skip_flag (keys %$skip_hash) {
+		my $skip_applets = $skip_hash->{$skip_flag};
+		foreach my $skip_applet (@$skip_applets) {
+			if ($skip_applet eq $applet) {
+				push @newdefault, "n if $skip_flag";
+				break
+			}
+		}
+	}
+	push @newdefault, $default;
+
+	return \@newdefault;
+}
+
+sub skip_applet_if_prefer_full($$) {
+	my $applet = shift;
+	my $default = shift;
+
+	my %skip_hash = (
+	   "BUSYBOX_USE_BRIDGE", [ "BRCTL" ],
```

Comment:
The one issue I can with that (otherwise I think something like that could work, although I don't think the metadata would be enough; you'd need to scan install sections I think) is that it would require the tree to have the various replacement packages installed from feeds (for ones from packages).